pkg/templates/latency.go

```diff
@@ -20,6 +20,33 @@ var (
 	)
 )
 
+// This map contains a mapping between request classes and the latency buckets
+// we use in Prometheus. 
+// 
+// It assumes that the observation metric is an histogram where the following
+// buckets exist: 0.1, 0.2, 0.5, 1, 2, 5 and 10.
+//
+// The bucket values were chosen based on the performance of payments-service
+// over a  month. The spreadsheet linked below includes the latency percentiles
+// of payments-service for each one of the routes over a month. We chose the
+// bucket values based on keeping in mind that:
+// - We wanted to choose a set of buckets we could use across our 95th and 99th
+//   latency percentiles
+// - The buckets were generic and had an even distribution of routes, i.e. all
+//   buckets apply to an similar number of routes so that there are no special
+//   cases
+//
+// https://docs.google.com/spreadsheets/d/1CAe6gpgZdjYjBy44bdYSBD0ZA8yxSkMkiqjLiZ_8NRo/edit#gid=15427288
+var requestClassToLatencyBucket = map[string]string{
+  "fast++": "0.1", // 100 ms
+  "fast+": "0.2",  // 200 ms
+  "fast": "0.5",   // 500 ms
+  "ok": "1",       // 1s
+  "slow": "2",     // 2s
+  "slow+": "5",    // 5s
+  "slow++": "10",  // 10s
```

Comment:
Names were chosen without much thinking. Open to suggestions.

Reply:
The values we chose match the buckets we already have in payments-service, so no need to create new buckets (which is nice because if we had to create new buckets all the metrics we have would have no meaning). These buckets turn out to fulfil both conditions in the comment above.